devtools/common/src/main/java/io/quarkus/templates/rest/BasicRest.java

```diff
@@ -11,97 +11,79 @@
+
         createIndexPage();
         createDockerFiles();
         createDockerIgnore();
         createApplicationConfig();
     }
 
     private void setupContext() {
-        MojoUtils.getAllProperties().forEach((k, v) -> context.put(k.replace("-", "_"), v));
-
-        if (className != null) {
+        if (context.get(CLASS_NAME) != null) {
             String packageName = (String) context.get(PACKAGE_NAME);
 
-            if (className.endsWith(MojoUtils.JAVA_EXTENSION)) {
-                className = className.substring(0, className.length() - MojoUtils.JAVA_EXTENSION.length());
-            } else if (className.endsWith(MojoUtils.KOTLIN_EXTENSION)) {
-                className = className.substring(0, className.length() - MojoUtils.KOTLIN_EXTENSION.length());
-            }
-
-            if (className.contains(".")) {
-                int idx = className.lastIndexOf('.');
-                packageName = className.substring(0, idx);
-                className = className.substring(idx + 1);
-            }
-
             if (packageName != null) {
                 File packageDir = new File(srcMain, packageName.replace('.', '/'));
                 File testPackageDir = new File(testMain, packageName.replace('.', '/'));
                 srcMain = mkdirs(packageDir);
                 testMain = mkdirs(testPackageDir);
             }
-
-            context.put(CLASS_NAME, className);
-            context.put(RESOURCE_PATH, path);
-
-            if (packageName != null) {
-                context.put(PACKAGE_NAME, packageName);
-            }
         }
     }
 
     private void createClasses() throws IOException {
-        File classFile = new File(srcMain, className + getProperSourceExtension());
-        File testClassFile = new File(testMain, className + "Test" + getProperSourceExtension());
-        File itTestClassFile = new File(testMain, "Native" + className + "IT" + getProperSourceExtension());
-        generate(
-                getSourceType() == SourceType.JAVA
-                        ? "templates/resource-template.ftl"
-                        : "templates/resource-template-kotlin.ftl",
-                context, classFile, "resource code");
-        generate(
-                getSourceType() == SourceType.JAVA
-                        ? "templates/test-resource-template.ftl"
-                        : "templates/test-resource-template-kotlin.ftl",
-                context, testClassFile, "test code");
-        generate(
-                getSourceType() == SourceType.JAVA
-                        ? "templates/native-test-resource-template.ftl"
-                        : "templates/native-test-resource-template-kotlin.ftl",
-                context, itTestClassFile, "IT code");
+        final Object className = context.get(CLASS_NAME);
+        // If className is null we disable the generation of the Jax-RS resource.
+        if (className != null) {
+            final String extension = type.getExtension();
```

Comment:
Any reason why some variables are final and some are not? 

I don't think it adds anything there.

Reply:
Force of habit. I make everything final when I can but I agree that it doesn't add anything (apart from extra words) :)